pkg/scheduler/framework/runtime/framework.go

```diff
@@ -346,12 +330,39 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, opts ...Opti
 		fillEventToPluginMap(p, options.clusterEventMap)
 	}
 
+	// initialize plugins per individual extension points
 	for _, e := range f.getExtensionPoints(profile.Plugins) {
 		if err := updatePluginList(e.slicePtr, *e.plugins, pluginsMap); err != nil {
 			return nil, err
 		}
 	}
 
+	// initialize multiPoint plugins to their expanded extension points
+	if len(profile.Plugins.MultiPoint.Enabled) > 0 {
+		if err := f.expandMultiPointPlugins(profile, pluginsMap); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(f.queueSortPlugins) != 1 {
+		return nil, fmt.Errorf("one queue sort plugin required for profile with scheduler name %q", profile.SchedulerName)
+	}
+
+	var totalPriority int64
+	if len(profile.Plugins.MultiPoint.Enabled) > 0 {
+		err, multiPointWeights := getScoreWeights(f, totalPriority, profile.Plugins.MultiPoint.Enabled)
+		if err != nil {
+			return nil, err
+		}
+		totalPriority += multiPointWeights
```

Comment:
The reasoning here is that we're still doing checking in this function to make sure the total score doesn't exceed the max value. Since we could have different score plugins configured in both MultiPoint and Score, we need to consider both

Reply:
wouldn't that consider the weight for the same plugin twice if it was overridden?

Reply:
True, so we could handle that by passing another set between the two calls. I'll add that.

Unfortunately we can't just use the final expanded weights without refactoring how the framework stores Score plugins. That's not a huge refactor, but maybe a better todo

Reply:
@ahg-g okay, solved this with the changes in https://github.com/kubernetes/kubernetes/pull/105611/commits/a517726dae02b9a82073391489f0e7d7b1eaa8a9

The pluginWeights are a map, so we don't need to worry about double-calculation. Combine it into one call by appending MultiPoint plugins to Score plugins. If the plugin already exists, assume that's because it was present earlier in the list as a Score plugin (don't know how we want to handle specifying the same plugin multiple times with different weights... I think that's user error). Check that each plugin actually implements `Score` before calculating the weight.

Reply:
thanks, you can resolve this comment.